lte/cloud/go/services/lte/obsidian/models/conversion.go

```diff
@@ -381,6 +381,46 @@ func (m *GatewayNonEpsConfigs) ToUpdateCriteria(networkID string, gatewayID stri
 	return cellularConfig.ToUpdateCriteria(networkID, gatewayID)
 }
 
+func (m *GatewayDNSConfigs) FromBackendModels(networkID string, gatewayID string) error {
+	cellularConfig := &GatewayCellularConfigs{}
+	err := cellularConfig.FromBackendModels(networkID, gatewayID)
+	if err != nil {
+		return err
+	}
+	*m = *cellularConfig.DNS
+	return nil
+}
+
+func (m *GatewayDNSConfigs) ToUpdateCriteria(networkID string, gatewayID string) ([]configurator.EntityUpdateCriteria, error) {
+	cellularConfig := &GatewayCellularConfigs{}
+	err := cellularConfig.FromBackendModels(networkID, gatewayID)
+	if err != nil {
+		return nil, err
+	}
+	cellularConfig.DNS = m
+	return cellularConfig.ToUpdateCriteria(networkID, gatewayID)
+}
+
+func (m *GatewayDNSRecords) FromBackendModels(networkID string, gatewayID string) error {
+	cellularConfig := &GatewayCellularConfigs{}
+	err := cellularConfig.FromBackendModels(networkID, gatewayID)
+	if err != nil {
+		return err
+	}
+	*m = cellularConfig.DNS.Records
```

Comment:
nilcheck here as well, cellularConfig.DNS can be nil since the field is optional